### PR TITLE
Modified project details - role based users in projects

### PR DESCRIPTION
**Pull request by Pearl1594**
Added details with respect to support for:
- multiple project admins
- project roles to restrict users from performing operations at project level

### CODE REVIEW

**Comment by Pearl1594** [comment]
requesting docbuild

**Comment by Pearl1594** [comment]
requesting docbuild

**Comment by Pearl1594** [comment]
requesting docbuild

File: source/adminguide/projects.rst [hunk 1]
```diff
@@ -280,36 +296,21 @@ feature is enabled in the cloud as described in `“Setting
...
-
-#. Type the user name or email address of the new member you want to
-   add, and click Invite. Type the CloudStack user name if you chose
-   Account in the previous step. If you chose Email, type the email
-   address. You can invite only people who have an account in this cloud
-   within the same domain as the project. However, you can send the
-   invitation to any email address.
-
-#. To view and manage the invitations you have sent, return to this tab.
+5. To view and manage the invitations you have sent, return to this tab.
```
Comment (DaanHoogland) [inline source/adminguide/projects.rst:313]: should th `5.` here be a `#.`?

File: source/adminguide/projects.rst [hunk 2]
```diff
@@ -32,17 +32,33 @@ You can configure CloudStack to allow any user to create a new project,
...
+operations (or APIs) by associating a project-level role to the
+user or account.
+
+**NOTE:** Project Roles work over Account level Roles. If a user/account is
+added to a project without a project role, it would imply that the
+user / account added will have access to all APIs that are made available
+by the Account level role. If there are no specific deny rules in the
+project role, it would again fallback onto the account-level role to decide
+whether the user has permissions to perform a specific action.
+
```
Comment (DaanHoogland) [inline source/adminguide/projects.rst:57]: i would like to see a second note to make clear that a user's rights can not be widened in comparison to their account role but only restricted.

File: source/adminguide/projects.rst (hunk 1 above)
Comment (Pearl1594) [reply source/adminguide/projects.rst]: done

File: source/adminguide/projects.rst (hunk 2 above)
Comment (Pearl1594) [reply source/adminguide/projects.rst:61]: Added relevant details

**Comment by Pearl1594** [comment]
requesting docbuild

**Comment by Pearl1594** [comment]
requesting docbuild

File: source/adminguide/projects.rst [hunk 3]
```diff
@@ -32,17 +32,38 @@ You can configure CloudStack to allow any user to create a new project,
...
+such as project VMs, fellow project members, project-related alerts, and so on.
+
+From CloudStack 4.15 onwards, it is possible for a project to have
+multiple project administrators and to add/invite specific users of
+an account to a project in addition to adding accounts. By means of
+Project Roles associated with a user or an account of the project,
+it is possible to restrict access of users in a project, i.e., in
+addition to account-level roles, one can further restrict access to
+operations (or APIs) by associating a project-level role to the
+user or account. However, If an account has already been added, one will not
```
Comment (GabrielBrascher) [inline source/adminguide/projects.rst:49]: `However, If` :arrow_right: `However, if`

File: source/adminguide/projects.rst [hunk 4]
```diff
@@ -32,17 +32,38 @@ You can configure CloudStack to allow any user to create a new project,
...
+be able to associate a role a specific user of that account.
+
+**NOTE:** Project Roles work over Account level Roles. If a user/account is
+added to a project without a project role, it would imply that the
+user / account added will have access to all APIs that are made available
+by the Account level role. If there are no specific deny rules in the
+project role, it would again fallback onto the account-level role to decide
+whether the user has permissions to perform a specific action. It is also to be
+noted that Project roles are restricictive in nature, i.e., to say that, one may
```
Comment (GabrielBrascher) [inline source/adminguide/projects.rst:58]: restricictive :arrow_right: restrictive

File: source/adminguide/projects.rst [hunk 5]
```diff
@@ -32,17 +32,38 @@ You can configure CloudStack to allow any user to create a new project,
...
+not allow a user to perform an operation that is NOT allowed at Account level.
+Even if a rule is added at the project level, allowing such an action, it will not
+have any effect as the action will be prohibitted by the Account Role.
```
Comment (GabrielBrascher) [inline source/adminguide/projects.rst:61]: prohibitted :arrow_right: prohibited

File: source/adminguide/projects.rst [hunk 6]
```diff
@@ -32,17 +32,38 @@ You can configure CloudStack to allow any user to create a new project,
...
+addition to account-level roles, one can further restrict access to
+operations (or APIs) by associating a project-level role to the
+user or account. However, If an account has already been added, one will not
+be able to associate a role a specific user of that account.
+
+**NOTE:** Project Roles work over Account level Roles. If a user/account is
+added to a project without a project role, it would imply that the
+user / account added will have access to all APIs that are made available
+by the Account level role. If there are no specific deny rules in the
+project role, it would again fallback onto the account-level role to decide
```
Comment (GabrielBrascher) [inline source/adminguide/projects.rst:56]: fallback :arrow_right: fall back

File: source/adminguide/projects.rst [hunk 7]
```diff
@@ -32,17 +32,38 @@ You can configure CloudStack to allow any user to create a new project,
...
+whether the user has permissions to perform a specific action. It is also to be
+noted that Project roles are restricictive in nature, i.e., to say that, one may
+not allow a user to perform an operation that is NOT allowed at Account level.
```
Comment (GabrielBrascher) [inline source/adminguide/projects.rst:59]: NOT allowed at Account level :arrow_right: NOT allowed at **the** Account level

**Comment by GabrielBrascher** [comment]
@Pearl1594 @DaanHoogland Is this still draft or can we put it into "Ready for review"?

**Comment by Pearl1594** [comment]
@GabrielBrascher I've addressed your comments. Thanks for pointing them out.

**Comment by Pearl1594** [comment]
requesting docbuild

**Pull request merged**